src/v/kafka/server/handlers/topics/types.cc

```diff
@@ -85,7 +85,7 @@ get_shadow_indexing_mode(const config_map_t& config) {
     if (si_enabled && *si_enabled) {
         mode = mode == model::shadow_indexing_mode::archival
                  ? model::shadow_indexing_mode::full
-                 : mode = model::shadow_indexing_mode::fetch;
+                 : model::shadow_indexing_mode::fetch;
```

Comment:
woah nice catch. did clang-tidy warn here?

Reply:
no, it was not clang-tidy. It was GCC-12. See the cover letter of this PR. I was trying to build Redpanda with GCC.